strawberry/experimental/pydantic/v2_compat.py

```diff
@@ -0,0 +1,104 @@
+import dataclasses
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Callable, Dict, Optional, Type
+
+import pydantic
+from pydantic import BaseModel
+from pydantic.version import VERSION as PYDANTIC_VERSION
+
+if TYPE_CHECKING:
+    from pydantic.fields import FieldInfo
+
+IS_PYDANTIC_V2: bool = PYDANTIC_VERSION.startswith("2.")
+IS_PYDANTIC_V1: bool = not IS_PYDANTIC_V2
```

Comment:
```suggestion
IS_PYDANTIC_V2: bool = PYDANTIC_VERSION.startswith("2.")
IS_PYDANTIC_V1: bool = PYDANTIC_VERSION.startswith("1.")
```

we might have this code until v3 :P